### PR TITLE
fix(low): check if points are on the curve before addition

### PR DESCRIPTION
**Pull request by heueristik**
**Title**: `Missing validation of input points to ensure coordinate
bounds and curve membership`
**Type**: `Implementation`
**Severity**: `Low`
**Exploitability**: `Low`
**Impact**: `Medium`

### CODE REVIEW

File: contracts/test/libs/EllipticCurve.t.sol [hunk 1]
```diff
@@ -0,0 +1,270 @@
...
+ * @dev Property-based tests for EllipticCurve.ecAdd using Foundry fuzzing.
+ *
+ * These tests verify GROUP PROPERTIES with points VERIFIED to be on the curve.
+ * Points are generated via scalar multiplication: P = k × G
+ * This guarantees all test points satisfy the curve equation y² = x³ + ax + b
+ *
+ * Run with: forge test -vv
+ * Run with more fuzz runs: forge test --fuzz-runs 10000 -vv
+ */
+contract EllipticCurvePropertiesTest is Test {
```
Comment (agureev) [inline contracts/test/libs/EllipticCurve.t.sol]: Wanted to record this here: we should add two fuzzing tests in the future, one to test zero-addition to arbitrary points and the other to test that `add` actually produces an on-curve point given two on-curve points

File: bindings/src/conversion.rs [hunk 2]
```diff
@@ -7,7 +7,7 @@ use arm_risc0::logic_instance::{AppData, ExpirableBlob};
 use arm_risc0::logic_proof::LogicVerifierInputs;
 use arm_risc0::proving_system::encode_seal;
 
-use arm_risc0::transaction::{Delta, Transaction};
+use arm_risc0::transaction::{Delta as ArmDelta, Transaction};
```
Comment (agureev) [inline bindings/src/conversion.rs]: Why do we want this change in this PR?

File: contracts/src/libs/proving/Delta.sol [hunk 3]
```diff
@@ -20,15 +20,27 @@ library Delta {
...
+    uint256 internal constant _BB = 7;
+
+    /// @notice The field prime modulus (2^256 - 2^32 - 977) of the secp256k1 (K-256) elliptic curve (y² = x³ + ax + b).
     uint256 internal constant _PP = 0xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFEFFFFFC2F;
 
     /// @notice Thrown if the recovered delta public key doesn't match the delta instance.
     error DeltaMismatch(address expected, address actual);
 
+    /// @notice Thrown when a provided point is not on the curve.
+    error PointNotOnCurve(CurvePoint point);
```
Comment (agureev) [inline contracts/src/libs/proving/Delta.sol]: Let's add a test testing this revert

File: contracts/test/libs/EllipticCurve.t.sol (hunk 1 above)
Comment (heueristik) [reply contracts/test/libs/EllipticCurve.t.sol]: Added in fe5cdbe95cf73ee6a566fd6b17ce9e2f02c56c87

File: contracts/src/libs/proving/Delta.sol (hunk 3 above)
Comment (heueristik) [reply contracts/src/libs/proving/Delta.sol]: Added in fe5cdbe95cf73ee6a566fd6b17ce9e2f02c56c87

File: contracts/src/libs/proving/Delta.sol [hunk 4]
```diff
@@ -10,43 +10,60 @@ import {EfficientHashLib} from "@solady/utils/EfficientHashLib.sol";
...
-    /// @notice Adds two elliptic curve points and returns the resulting value.
-    /// @param p1 The first curve point.
-    /// @param p2 The second curve point.
+    /// @notice Adds two delta points and returns the sum.
+    /// @param lhs The left-hand side point that can also be the zero delta.
+    /// @param rhs The right-hand side point that must be a curve point.
     /// @return sum The resulting curve point.
-    function add(CurvePoint memory p1, CurvePoint memory p2) internal pure returns (CurvePoint memory sum) {
-        (sum.x, sum.y) = EllipticCurve.ecAdd({_x1: p1.x, _y1: p1.y, _x2: p2.x, _y2: p2.y, _aa: _AA, _pp: _PP});
+    /// @dev Note that only the right-hand side point is checked to allow adding the zero delta from the left.
```
Comment (agureev) [inline contracts/src/libs/proving/Delta.sol]:
```suggestion
    /// @dev Note that only the right-hand side point is checked to allow adding the zero delta from the left. 
    /// @dev This is done due to the delta points being added sequentially starting from the zero delta in our implementation.
```

File: contracts/test/proofs/DeltaProof.t.sol [hunk 5]
```diff
@@ -302,12 +303,70 @@ contract DeltaProofTest is Test {
...
+        );
+
+        // Add the two points and check that the sum is a curve point.
+        Delta.Point memory sum = lhs.add(rhs);
+        assertTrue(
+            EllipticCurve.isOnCurve({_x: sum.x, _y: sum.y, _aa: Delta._AA, _bb: Delta._BB, _pp: Delta._PP}),
+            "Sum must be on the curve."
+        );
+    }
+
```
Comment (agureev) [inline contracts/test/proofs/DeltaProof.t.sol]: As mentioned before, let's add a test such that the zero point + other point is the other point

File: contracts/test/libs/EllipticCurve.t.sol (hunk 1 above)
Comment (agureev) [reply contracts/test/libs/EllipticCurve.t.sol]: Commented regarding one of the test, namely

> one to test zero-addition to arbitrary points

File: contracts/src/libs/proving/Delta.sol [hunk 6]
```diff
@@ -40,6 +52,10 @@ library Delta {
     /// @param p2 The second curve point.
     /// @return sum The resulting curve point.
```
[the remark this answers is not in the document]
Comment (agureev) [reply contracts/src/libs/proving/Delta.sol]: Left comments on the dev string

File: contracts/test/libs/EllipticCurve.t.sol (hunk 1 above)
Comment (heueristik) [reply contracts/test/libs/EllipticCurve.t.sol]: Argh, good catch. I have added two asserts, but then probably lost them in a git stash. Here we go 6d7c7ac03dfe0f74c5f0338e2471344843001591
